### DISCUSSION TITLE
How documentation is generated?

### DISCUSSION THREAD

**Discussion by richardtop** [SwiftPackageIndex/SwiftPackageIndex-Server#2736] [Q&A] [answered]
Hi, I'm interested in how does the documentation page get generated:
e.g. for my repo: https://swiftpackageindex.com/richardtop/calendarkit/1.1.9/documentation/calendarkit/alldayview

What is the tooling used to generate the html pages? Is it some standard docc server or something more customized? Does it even run on docc?

I'm interested in the documentation solution that transforms docc archive into static html files (not the files required to be served by the web server, as with the default Apple docc design)

**Comment by finestructure** [comment]
Hi Richard, we're essentially following the [standard procedure for static hosting of DocC generated docs](https://swiftpackageindex.com/apple/swift-docc-plugin/1.3.0/documentation/swiftdoccplugin/generating-documentation-for-hosting-online). There might also be some additional info in this WWDC video: https://developer.apple.com/videos/play/wwdc2021/10236/

Hope that helps!

**Reply by richardtop** [reply]
Does that still require setting up a server? Or can I just open the index.html and view it locally with that approach?

**Reply by finestructure** [reply, marked as answer]
I'm pretty sure it'll require a server serving files from the directory but you can launch one via the preview command:

```
swift package --disable-sandbox preview-documentation -p 8080 --target $target
```

**Reply by richardtop** [reply]
Clear, thanks. I'm looking for serverless approach, similar to this product: https://github.com/DoccZz/docc2html, however it still doesn't cut it as not very reliable.

**Reply by heckj** [reply]
Hi Richard - jumping in, although not one of the SPI team members. The closest you'll find to anything that generates static HTML output from a DocC collection today (converting a Docc archive into static HTML pages) is Helge's project you listed above. You're not the only one who wants that kind of capability, but so far nobody's really stepped up to work through all the details and possible templating, as well as handling the variety of outputs that DocC includes. I spoke with Helge just a few weeks ago, and that project will happily take updates if you're inclined to push it a bit further down the road.

**Reply by richardtop** [reply]
I've ended up moving with Jazzy https://github.com/realm/jazzy/ for my use-case. It generates exactly what I wanted (static html website, no need to run http server).
